e2e/src/componentsController.spec.ts

```diff
@@ -636,4 +646,462 @@ describe('Components Controller', () => {
+              description: 'updated-description-with-invalid-id',
+            },
+          }
+        );
+
+      expect(updateComponentResponse).rejects.toThrow();
+
+      await updateComponentResponse.catch((reason) => {
+        expect(reason.statusCode).toBe(404);
+      });
+    });
+    it('should not update product family component with invalid credentials', async () => {
+      const newProductPayload = {
+        product: {
+          name: `update-product-family-component-with-invalid-credentials-${uid()}`,
+          handle: `update-product-family-component-with-invalid-credentials-${uid()}`,
+          description: '',
+          requireCredit_card: true,
+          priceInCents: BigInt(1000),
+          interval: 1,
+          intervalUnit: IntervalUnit.Month,
+        },
+      };
+      const productFamilyId = productFamily?.id || 0;
+
+      const meteredComponent: CreateMeteredComponent = {
+        meteredComponent: {
+          productFamilyId,
+          name: `update-component-name${uid()}`,
+          description: 'test',
+          quantity: 1,
+          unitName: 'test',
+          price: 1,
+          handle: uid(),
+          pricingScheme: PricingScheme.PerUnit,
+          unitPrice: 1,
+        },
+      };
+
+      const { component } = await prepareDataForComponentResponse(
+        newProductPayload,
+        productFamilyId.toString(),
+        meteredComponent
+      );
+
+      const componentId = component.id?.toString() || '';
+
+      const updateComponentResponse =
+        invalidComponentsController.updateProductFamilyComponent(
+          productFamilyId,
+          componentId,
+          {
+            component: {
+              name: `updated-product-family-component-name-with-invalid-credentials-${uid()}`,
+              description: `updated-product-family-component-description-with-invalid-credentials-${uid()}`,
+            },
+          }
+        );
+
+      expect(updateComponentResponse).rejects.toThrow();
+
+      await updateComponentResponse.catch((reason) => {
+        expect(reason.statusCode).toBe(401);
+      });
+    });
+  });
+  describe('Promote Component Price Point to default', () => {
+    it('should promote component price to default', async () => {
+      const newProductPayload: CreateOrUpdateProductRequest = {
+        product: {
+          name: `promote-component-price-to-default-name-${uid()}`,
+          handle: `promote-component-price-to-default-handle-${uid()}`,
+          description: '',
+          requireCredit_card: true,
+          priceInCents: BigInt(1000),
+          interval: 1,
+          intervalUnit: IntervalUnit.Month,
+        },
+      };
+
+      const price: Price = {
+        starting_quantity: '1',
+        ending_quantity: '',
+        unit_price: '1',
+        startingQuantity: '',
+        unitPrice: '',
+      };
+
+      const productFamilyId = productFamily?.id || 0;
```

Comment:
I would rather cast it everywhere than do `or`. It will fail anyways if the productFamily is not there, but might lead to some unclear outcomes, options:

1.  Cast to proper value:

```
(productFamily.id as string)
```

It will fail if the `productFamily` is not there

2. Handle empty values with extra assertion

```
expect(productFamily).toBe({....})

...
```

3. Extra condition

```
if (!productFamily) { throw Error('Test data malformed') }

...
```